### ISSUE TITLE
Old copy paste bug 

### ISSUE THREAD

**Issue by Angry0Fellow** [xournalpp/xournalpp#5653]
### Operating System

Linux

### (Linux only) Distribution

Manjaro

### (Linux only) Desktop Environment

Gnoome

### (Linux Only) Display Server

Wayland

### Installation Method

pacman

### Xournal++ Version

1.2.2-1.2.3

### libgtk Version

3.24.41

### Bug Description

The bug is exactly the same [#2831](https://github.com/xournalpp/xournalpp/issues/2831). The origianl issue was closed but i'm still having the issue.

### Expected Behaviour

Copy and paste should not create a larger copy that cannot be erased.  

### Steps to Reproduce

1. copy any pen strokes
2.paste them anywhere

**Comment by rolandlo** [comment]
The fix went into version 1.2.0 and you have only tried versions 1.1.2-1.1.3 (which are relatively old). No wonder you still encounter the issue. Please upgrade.

**Comment by Angry0Fellow** [comment]
[Uploading Screencast from 2024-05-02 09-14-49.webm…]()

**Comment by Angry0Fellow** [comment]
As a workaround you can copy the exact same strokes twice and it will work as expected.
[Workaround.webm](https://github.com/xournalpp/xournalpp/assets/98543803/fc357541-51ee-4907-8b8f-4ac8aa260f92)
Just pasting the same strokes will still fail.

**Comment by Photon89** [comment]
I am hit by this issue for a few days, it worked flawlessly before that though. Using version 1.2.5, but it also happened with some git commit between 1.2.3 and 1.2.4. 

I think, I can pin it down to an upgrade which happened during the last ten days. Unfortunately, there was a rather huge update during this period of time: https://pastebin.com/0W5n69ei

**Comment by Photon89** [comment]
I can reproduce it with 1.2.6 on an Arch Linux machine. Since Arch and Manjaro both update early, I'd expect the issue to pop up in other distros after some period of time as well.

Also, I tried with a clean user profile (move .config/xournalpp) and the problem persists.

**Comment by tattsan** [comment]
I've also run into the similar issue on Arch Linux, but it is due to the clipboard application.
I'm using `xfce4-clipman-plugin` for a clipboard manager. After upgrading the app. to  1.6.7-1, the manager copies the X++  selection as a bitmap object, not  a pen stroke object.

**Comment by Photon89** [comment]
I removed the Clipman plugin from the Xfce Panel and verified that no process containing "clipman" is running, but unfortunately the issue persists for me.

**Comment by tattsan** [comment]
I've tested both `xournalpp-1.2.6-x86_64.AppImage` and `master` (build from source), on X11. 
Is this bug related to wayland?

In my environment, I can use X++ without any problems by downgrading `xfce4-clipman-plugin`.

**Comment by Photon89** [comment]
I am using X11 as well (I think, Xfce doesn't support Wayland so far). I downgraded the clipman plugin to 1.6.6 but it doesn't make a difference. Are you sure that it is just the clipman plugin and no further package?

**Comment by tattsan** [comment]
* I'm using xfce4-panel, not on xfce4 environment, but on xmonad.
* Currently, I'm using the latest xfce4-clipman-plugin with the following modification:

```
--- PKGBUILD.orig       2024-12-30 14:00:17.351900664 +0900
+++ PKGBUILD    2024-12-31 12:39:01.310784290 +0900
@@ -5,7 +5,7 @@

 pkgname=xfce4-clipman-plugin
 pkgver=1.6.7
-pkgrel=1
+pkgrel=101
 pkgdesc="A clipboard plugin for the Xfce4 panel"
 arch=('x86_64')
 license=('GPL-2.0-or-later')
@@ -20,6 +20,11 @@

 prepare() {
   cd $pkgname
+  git revert -n a7e5c15ae41e6d89d696769a9f46bc7b41fccfb6
+  git revert -n eb944fb2c5ec5bdf5b056d6989345ba37d662c1a
+  git revert -n 537e212e0aa45acc1fb2d7419bff684f5ad2e394
+  git revert -n d324ec82a878d17051557edeb02eb77b6a21b21d
+  git revert -n dc41e8791db2824a91dc07f28eacc42e299f3632
   git submodule init
   git config submodule.mate-submodules.url "$srcdir/protocols/wlr-protocols"
   git -c protocol.file.allow=always submodule update
```

**Comment by Photon89** [comment]
I don't think that the clipman plugin is involved, at least in my case, because the issue appears even if it isn't running at all....

**Comment by Photon89** [comment]
Which libraries could be involved in the copy&paste process? I could try downgrading some to find the culprit.

**Comment by Photon89** [comment]
Looks like it is fixed in latest master!

**Comment by bhennion** [comment]
> Looks like it is fixed in latest master!

I don't think we did anything on our end, so I guess something has been updated in your system.

> Which libraries could be involved in the copy&paste process? I could try downgrading some to find the culprit.

Wayland or Xlib + GDK + GTK + some libraries related to your desktop environment and how it handles the clipboard. So a lot.

**Comment by Photon89** [comment]
Oh well. For me the issue remained fixed. @Taza53 might want to compare their update log with mine which I posted in https://github.com/xournalpp/xournalpp/issues/5653#issuecomment-2651551196

**Comment by youssefadly237** [comment]
relating to #7219 and this issue I had similar experience, and it was because of the clipboard if this is relevant in #6743

**Comment by domenicozambella** [comment]
I had the same issue ( see [#7219](https://github.com/xournalpp/xournalpp/issues/7219) ). 

I just realized that it depends on the setting of the clipboard. The issue appears with this settings:

<img width="854" height="729" alt="Image" src="https://github.com/user-attachments/assets/f212ce1e-10db-4e97-872d-dbcbac970cef" />

But it disappears when I change the setting to:

<img width="782" height="186" alt="Image" src="https://github.com/user-attachments/assets/c6f8ecc3-a707-48c3-b582-2aa63b72eb73" />
